source/Calamari.Common/Plumbing/Deployment/PackageRetention/CaseInsensitiveTinyType.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Dynamic;
+using YamlDotNet.Core.Tokens;
+
+namespace Calamari.Deployment.PackageRetention
+{
+    //TODO: Replace this basic tiny types implementation with the Octopus one.
```

Comment:
We can't replace this without modifying the TinyTypes project to support Framework 4.0.

Reply:
Bit of a dumb question - could we move Calamari to .NET 5? I'm guessing it would be pretty difficult

Reply:
We can, but AFAIK we also need to support older frameworks for customers who haven't updated yet.